audioplayer.cpp

```diff
@@ -0,0 +1,99 @@
+#include "audioplayer.h"
+
+vector<string> getFileNames(const string& folder) {
+    vector<string> names;
+    DIR* directory = opendir(folder.c_str());
+
+    if (directory == nullptr) {
+        abort();
+    }
+
+    struct dirent* ent;
+    while ((ent = readdir(directory)) != nullptr) {
+        if ((strcmp(ent->d_name, "..")) && (strcmp(ent->d_name, "."))) {
+            names.emplace_back(ent->d_name);
+        }
+    }
+    closedir(directory);
+    return names;
+}
+
+void AudioPlayer::closeAudio() {
+    callMciSendString("close mp3");
+}
+
+void AudioPlayer::playAudio() {
+    // save current volume
+    callMciSendString(("setAudio mp3 volume to " + currentVolume));
```

Comment:
Is setting volume to `currentVolume` necessary? Looks like we can only do it in `AudioPlayer::changeVolume`, no?

Reply:
> Is setting volume to `currentVolume` necessary? Looks like we can only do it in `AudioPlayer::changeVolume`, no?

I think no. I've tried to do like you wrote. I doesen't apply current volume level after restarting, pausing, etc.